src/readbin/compare.ml

```diff
@@ -0,0 +1,119 @@
+open Core_kernel.Std
+open Cmdliner
+open Bap.Std
+open Bap_plugins.Std
+
+exception No_unstripped_file
+
+let print_source = function
+  | `bw -> print_endline "bw"
+  | _ -> print_endline "others"
+
+let source = [
+  "BW", `bw;
+  "SymTbl", `symtbl;
+  "User", `user;
+  "Ida", `ida;
+]
+
+let tool : _ Term.t =
+  let doc = "The tool that we are going to evaluate. It can be BW or Ida." in
```

Comment:
No, it can be either `BW, SymTbl, User, Ida`, if it is not, the refactor your types to reflect this. Moreover, there is `doc_alts_enum` function, use it to enumerate possibilities